Special props translation (#28)
reactjs/pt-BR.reactjs.org: content/warnings/special-props.md

permalink: warnings/special-props.html
---

> A maioria das propriedades de um elemento JSX são passadas no componente, porém, existem duas propriedades especiais (`ref` e `key`) que são usadas pelo React, e portanto não são passadas para um componente.

Reviewer: ```suggestion
A maioria das propriedades de um elemento JSX são passadas no componente. Porém, existem duas propriedades especiais (`ref` e `key`) que são usadas pelo React, e portanto não são passadas para um componente.
```

> A maioria das propriedades de um elemento JSX são passadas no componente. Porém, existem duas propriedades especiais (`ref` e `key`) que são usadas pelo React, e portanto não são passadas para um componente.

Reviewer 2: `... passadas no componente. ...`
`... e portanto não são passadas para um componente ...`

Acho que ficaria mais coerente se usassemos `para` para os dois casos.

> Por exemplo, ao tentar acessar `this.props.key` de um componente (ex: a função de renderizar ou [propTypes](/docs/typechecking-with-proptypes.html#proptypes)) não é definido. Caso você precise acessar o mesmo valor de dentro de um componente filho, você deve passá-lo como uma propriedade diferente (ex: `<ListItemWrapper key={result.id} id={result.id} />`). Por mais que isso possa parecer redundante é importante separar a lógica das dicas de reconciliação da aplicação.

Reviewer 2: `Por mais que isso possa parecer redundante [,] é imp...`
Faltou a virgula

Para mim tudo bem, mas parece confusa a frase `é importante separar a lógica das dicas de reconciliação da aplicação.`
seria menos estranhos aos meus ouvidos: `é importante separar a lógica das dicas de reconciliação da [lógica] da aplicação.`

> Por exemplo, ao tentar acessar `this.props.key` de um componente (ex: a função de renderizar ou [propTypes](/docs/typechecking-with-proptypes.html#proptypes)) não é definido. Caso você precise acessar o mesmo valor de dentro de um componente filho, você deve passá-lo como uma propriedade diferente (ex: `<ListItemWrapper key={result.id} id={result.id} />`). Por mais que isso possa parecer redundante, é importante separar a lógica das dicas de reconciliação da aplicação.

Reviewer: ```suggestion
Por exemplo, ao tentar acessar `this.props.key` de um componente (ex: a função de renderizar ou [propTypes](/docs/typechecking-with-proptypes.html#proptypes)) não é definido. Caso você precise acessar o mesmo valor de dentro de um componente filho, você deve passá-lo como uma propriedade diferente (ex: `<ListItemWrapper key={result.id} id={result.id} />`). Por mais que isso possa parecer redundante é importante separar a lógica das dicas de reconciliação da aplicação.
```

> Por exemplo, ao tentar acessar `this.props.key` de um componente (ex: a função de renderizar ou [propTypes](/docs/typechecking-with-proptypes.html#proptypes)) não é definido. Caso você precise acessar o mesmo valor de dentro de um componente filho, você deve passá-lo como uma propriedade diferente (ex: `<ListItemWrapper key={result.id} id={result.id} />`). Por mais que isso possa parecer redundante, é importante separar a lógica das dicas de reconciliação da lógica aplicação.

Reviewer 3: ```suggestion
Por exemplo, ao tentar acessar `this.props.key` de um componente (ex: a função de renderizar ou [propTypes](/docs/typechecking-with-proptypes.html#proptypes)) essa prop não estará definida. Caso você precise acessar o mesmo valor de dentro de um componente filho, você deve passá-lo como uma prop diferente (ex: `<ListItemWrapper key={result.id} id={result.id} />`). Por mais que isso possa parecer redundante, é importante separar a lógica da aplicação das dicas de reconciliação.
```
